test/spherical_generator.cxx

```diff
@@ -0,0 +1,327 @@
+        RadialQuad::MuraKnowles, 99, 1.0,
+        {
+          {0 , 34, AngularQuad::Delley, 26},
+          {34, 50, AngularQuad::Delley, 50},
+          {50, 99, AngularQuad::Delley, 302},
+        }
+      };
+    }
+
+    REQUIRE(pruning_spec == ref_pruning_spec);
+  }
+
+  SECTION("Lebedev-Laikov") {
+
+    UnprunedSphericalGridSpecification unp{
+      RadialQuad::MuraKnowles, 99, 1.0,
+      AngularQuad::LebedevLaikov, 302
+    };
+
+    PrunedSphericalGridSpecification pruning_spec, ref_pruning_spec;
+
+    SECTION("Robust") {
+      pruning_spec = robust_psi4_pruning_scheme(unp);
+      ref_pruning_spec = PrunedSphericalGridSpecification{
+        RadialQuad::MuraKnowles, 99, 1.0,
+        {
+          {0 , 25, AngularQuad::LebedevLaikov, 26},
+          {25, 50, AngularQuad::LebedevLaikov, 194},
+          {50, 99, AngularQuad::LebedevLaikov, 302},
+        }
+      };
+    }
+
+    SECTION("Treutler") {
+      pruning_spec = treutler_pruning_scheme(unp);
+      ref_pruning_spec = PrunedSphericalGridSpecification{
+        RadialQuad::MuraKnowles, 99, 1.0,
+        {
+          {0 , 34, AngularQuad::LebedevLaikov, 26},
+          {34, 50, AngularQuad::LebedevLaikov, 50},
+          {50, 99, AngularQuad::LebedevLaikov, 302},
+        }
+      };
+    }
+
+    REQUIRE(pruning_spec == ref_pruning_spec);
+  }
+
+  SECTION("Womersley") {
+
+    UnprunedSphericalGridSpecification unp{
+      RadialQuad::MuraKnowles, 99, 1.0,
+      AngularQuad::Womersley, 339
+    };
+
+    PrunedSphericalGridSpecification pruning_spec, ref_pruning_spec;
+
+    SECTION("Robust") {
+      pruning_spec = robust_psi4_pruning_scheme(unp);
+      ref_pruning_spec = PrunedSphericalGridSpecification{
+        RadialQuad::MuraKnowles, 99, 1.0,
+        {
+          {0 , 25, AngularQuad::Womersley, 32},
+          {25, 50, AngularQuad::Womersley, 201},
+          {50, 99, AngularQuad::Womersley, 339},
+        }
+      };
+    }
+
+    SECTION("Treutler") {
+      pruning_spec = treutler_pruning_scheme(unp);
+      ref_pruning_spec = PrunedSphericalGridSpecification{
+        RadialQuad::MuraKnowles, 99, 1.0,
+        {
+          {0 , 34, AngularQuad::Womersley, 32},
+          {34, 50, AngularQuad::Womersley, 72},
+          {50, 99, AngularQuad::Womersley, 339},
+        }
+      };
+    }
+
+    REQUIRE(pruning_spec == ref_pruning_spec);
+  }
+
+}
+
+using bk_type  = IntegratorXX::Becke<double,double>;
+using mk_type  = IntegratorXX::MuraKnowles<double,double>;
+using mhl_type = IntegratorXX::MurrayHandyLaming<double,double>;
+using ta_type  = IntegratorXX::TreutlerAhlrichs<double,double>;
+
+using ah_type = IntegratorXX::AhrensBeylkin<double>;
+using de_type = IntegratorXX::Delley<double>;
+using ll_type = IntegratorXX::LebedevLaikov<double>;
+using wo_type = IntegratorXX::Womersley<double>;
+
+using sph_test_types = std::tuple<
+  std::tuple<bk_type, ah_type>,
+  std::tuple<bk_type, de_type>,
+  std::tuple<bk_type, ll_type>,
+  std::tuple<bk_type, wo_type>,
```

Comment:
If there was a list or dict of the supported radial and angular rules, this could just be replaced by looping over all possible combinations of the supported radial and angular rules.

Reply:
This is for the test generation (I'm testing that the runtime generator creates the same grid as the compile-time variants, which require a strong type), not meant to be interacted with in this manner. You can't make a list of strong types (... well, without `boost::mpl`).